R/dvhydrograph-render.R

```diff
@@ -3,199 +3,291 @@ dvhydrographPlot <- function(data) {
+#' relevant data, formats it, and then creates a DV Hydrogrpah
+#' plot from it.
+#' @param reportObject the full report JSON object
+createDVHydrographPlot <- function(reportObject){
+  #Rendering Options
   options(scipen=8)
-  
-  dvData <- parseDVData(data)
-  dvInfo <- dvData[['dvInfo']]
-  dvData <- dvData[['dvData']]
-  isInverted <- fetchReportMetadataField(data, 'isInverted')
-  
-  if (anyDataExist(dvData)) {
-    startDate <- flexibleTimeParse(data$reportMetadata$startDate, timezone=data$reportMetadata$timezone) 
-    endDate <- toEndOfDay(flexibleTimeParse(data$reportMetadata$endDate, timezone=data$reportMetadata$timezone))
+
+  #Validate Report Metadata
+  metaData <- fetchReportMetadata(reportObject)
+
+  requiredMetadataFields <- c(
+    'startDate',
+    'endDate',
+    'isInverted',
+    'timezone'
+  )
+
+  if(validateFetchedData(metaData, "metadata", requiredMetadataFields)){
+    #Get Necessary Report Metadata
+    timezone <- fetchReportMetadataField(reportObject, 'timezone')
+    excludeZeroNegativeFlag <- fetchReportMetadataField(reportObject, 'excludeZeroNegative')
+    excludeMinMaxFlag <- fetchReportMetadataField(reportObject, 'excludeMinMax')
+    invertedFlag <- fetchReportMetadataField(reportObject, 'isInverted')
+    startDate <- flexibleTimeParse(fetchReportMetadataField(reportObject, 'startDate'), timezone=timezone)
+    endDate <- toEndOfDay(flexibleTimeParse(fetchReportMetadataField(reportObject, 'endDate'), timezone=timezone))
     plotDates <- toStartOfDay(seq(startDate, endDate, by = 7 * 24 * 60 * 60))
-    
-    plot_object <- gsplot(ylog = dvInfo$logAxis, yaxs = 'i') %>%
+  }
+
+  #Get Basic Plot data
+  dvData <- list()
+
+  dvData[['stat1TimeSeries']] <- parseDVTimeSeries(reportObject, 'firstDownChain', 'downChainDescriptions1', timezone, excludeZeroNegativeFlag)
+  dvData[['stat1TimeSeriesEst']] <- parseDVTimeSeries(reportObject, 'firstDownChain', 'downChainDescriptions1', timezone, excludeZeroNegativeFlag, estimated=TRUE)
+  dvData[['stat2TimeSeries']] <- parseDVTimeSeries(reportObject, 'secondDownChain', 'downChainDescriptions2', timezone, excludeZeroNegativeFlag)
+  dvData[['stat2TimeSeriesEst']] <- parseDVTimeSeries(reportObject, 'secondDownChain', 'downChainDescriptions2', timezone, excludeZeroNegativeFlag, estimated=TRUE)
+  dvData[['stat3TimeSeries']] <- parseDVTimeSeries(reportObject, 'thirdDownChain', 'downChainDescriptions3', timezone, excludeZeroNegativeFlag)
+  dvData[['stat3TimeSeriesEst']] <- parseDVTimeSeries(reportObject, 'thirdDownChain', 'downChainDescriptions3', timezone, excludeZeroNegativeFlag, estimated=TRUE)
+  dvData[['comparisonTimeSeries']] <- parseDVTimeSeries(reportObject, 'comparisonSeries', 'comparisonSeriesDescriptions', timezone, excludeZeroNegativeFlag)
+  dvData[['comparisonTimeSeriesEst']] <- parseDVTimeSeries(reportObject, 'comparisonSeries', 'comparisonSeriesDescriptions', timezone, excludeZeroNegativeFlag, estimated=TRUE)
+  primaryPresentTS <- names(dvData)[1]
```

Comment:
add comment here - not sure I like the name using "Present". What does that mean?